.github/workflows/continuous-integration.yaml

```diff
@@ -631,24 +631,35 @@ jobs:
           name: py3-wheel
           path: dist/*.whl
       
-      # Run Pytest on a subset of our tests if the source code or our tests has changed, and if the 
-      # current workflow does no take place in a weekly or release CI. This step is also triggered
-      # if any dependency has been updated or if some changes were found in the conftest.py as well
-      # as the Makefile
+      # Run Pytest on a subset of our tests if :
+      # - the current workflow does no take place in a weekly or release CI
+      # - if the CI has been triggered manually (through GitHub's Action interface)
+      # - the source code has been changed
+      # - any tests utils (pytest, data) has been changed
+      # - any dependency has been updated 
+      # - conftest.py has been changed
+      # - Makefile has been changed
+      # If only some test files were changed, this step is skipped and each associated tests will be
+      # run individually in a following step (pytest_modified_tests_only)
       # If regular tests failed, a following script checks for flaky tests. If all failed tests 
       # are known flaky tests, they are re-run. Otherwise, the step exits with status 1.
       # The 'bash +e {0}' is added here in order to make sure that the step does not exit directly 
       # if 'make pytest' fails
       - name: PyTest Source Code (regular)
         id: pytest_regular
         if: |
-          steps.changed-files-in-pr.outcome == 'success' 
-          && (
-            steps.changed-files-in-pr.outputs.src_any_changed == 'true' 
-            || steps.changed-files-in-pr.outputs.tests_any_changed == 'true'
-            || steps.changed-files-in-pr.outputs.dependencies_any_changed == 'true'
-            || steps.changed-files-in-pr.outputs.conftest_any_changed == 'true'
-            || steps.changed-files-in-pr.outputs.makefile_any_changed == 'true'
+          (
+            (
+              steps.changed-files-in-pr.outcome == 'success' 
+              && (
+                steps.changed-files-in-pr.outputs.src_any_changed == 'true' 
+                || steps.changed-files-in-pr.outputs.tests_utils_any_changed == 'true'
+                || steps.changed-files-in-pr.outputs.dependencies_any_changed == 'true'
+                || steps.changed-files-in-pr.outputs.conftest_any_changed == 'true'
+                || steps.changed-files-in-pr.outputs.makefile_any_changed == 'true'
+              )
+            )
+            || fromJSON(env.IS_WORKFLOW_DISPATCH)
```

Comment:
regular tests should be run if CI is manually triggered

Reply:
What do you mean by manually triggered?

Reply:
through GitHub's [Action interface](https://github.com/zama-ai/concrete-ml/actions)

Reply:
meaning we manually trigger the workflow (it is not triggered automatically when opening a PR or during a weekly release for example)